lib/IrcHandlers/IrcActionHandler.js

```diff
@@ -0,0 +1,50 @@
+const Helpers = require("../Helpers.js");
+
+/**
+ * Handles the event when a user performs an action in the IRC channel.
+ */
+class IrcActionHandler {
+
+    // ---------------- Constructor ----------------
+
+    /**
+     * 
+     * @param {Array} ircBlackList - List of names to ignore. null/undefined to not ignore anyone.
+     * @param {boolean} enabled - If this handler enabled?
+     * @param {Function} action - The action to take when this handler is fired.
+     *                            Only parameter is a string, which is the message
+     *                            to send out.
+     */
+    constructor(ircBlackList, enabled, action) {
+        this._ircBlackList = ircBlackList;
+        this._action = action;
+        this.Enabled = enabled;
+    }
+
+    // ---------------- Functions ----------------
+
+    /**
+     * Constructs the message and fires this class's callback.
+     * @param {string} username - The user who performed the action.
+     * @param {string} channel - The channel the user has performed the action in.
+     * @param {string} userAction - The action the user performed.
+     */
+    ReportAction(username, channel, userAction) {
+        if (this.Enabled == false ) {
```

Comment:
You've got an extra space before the closing paren here.

Also, can we use a type-aware comparison here? `===` is generally preferable if the types should match. If this can be `===` I'd suggest a quick run-through of the other classes and see if the same can be done there.

Reply:
Went through all of the new handler classes and made them === instead of ==.